transport/processing/uploaders.py

```diff
@@ -0,0 +1,76 @@
+import ci.util
+
+
+class IdentityUploader():
+    def process(self, processing_job, target_as_source=False):
+        upload_request = processing_job.upload_request
+        if not target_as_source:
+            upload_request = processing_job.upload_request._replace(
+                target_ref=processing_job.upload_request.source_ref,
+            )
+        return processing_job._replace(upload_request=upload_request)
+
+
+class PrefixUploader():
+    def __init__(
+        self,
+        prefix,
+        context_url,
+        mangle=True,
+        **kwargs
+    ):
+        super().__init__(**kwargs)
```

Comment:
I think both the invocation of super-class's initialised, and the ```**kwargs`` parameter should be removed
also: consider using a dataclass (not 100% sure whether this is a good match in this case, though)?

Reply:
I've switched to a dataclass, I think the code looks better

Reply:
glad to hear you seem to like the change :-)